filament/backend/src/opengl/CocoaTouchExternalImage.mm

```diff
@@ -228,6 +229,21 @@ void main() {
     return texture;
 }
 
+CVOpenGLESTextureRef CocoaTouchExternalImage::createTextureFromImage(CVPixelBufferRef image, GLuint
+        glFormat, GLuint pixelFormat, size_t plane) noexcept {
+    const size_t width = CVPixelBufferGetWidthOfPlane(image, plane);
+    const size_t height = CVPixelBufferGetHeightOfPlane(image, plane);
+
+    CVOpenGLESTextureRef texture = nullptr;
+    UTILS_UNUSED_IN_RELEASE CVReturn success =
+            CVOpenGLESTextureCacheCreateTextureFromImage(kCFAllocatorDefault,
```

Comment:
According to [Apple's documentation](https://developer.apple.com/documentation/corevideo/1619860-cvopenglestexturecachecreatetext?language=objc), even for BGRA buffers `GL_RGBA` should be passed:

```
//Mapping a BGRA buffer as a source texture:
CVOpenGLESTextureCacheCreateTextureFromImage(kCFAllocatorDefault, textureCache, pixelBuffer, NULL, GL_TEXTURE_2D, GL_RGBA, width, height, GL_RGBA, GL_UNSIGNED_BYTE, 0, &outTexture);
```

Does changing it to `GL_BGRA` have an effect?

Reply:
Interesting, I wonder if there is a typo in their sample code there?

Reply:
> According to [Apple's documentation](https://developer.apple.com/documentation/corevideo/1619860-cvopenglestexturecachecreatetext?language=objc), even for BGRA buffers `GL_RGBA` should be passed:
> 
> ```
> //Mapping a BGRA buffer as a source texture:
> CVOpenGLESTextureCacheCreateTextureFromImage(kCFAllocatorDefault, textureCache, pixelBuffer, NULL, GL_TEXTURE_2D, GL_RGBA, width, height, GL_RGBA, GL_UNSIGNED_BYTE, 0, &outTexture);
> ```
> 
> Does changing it to `GL_BGRA` have an effect?

In my case, I decoded a video by AVAssetReader, and I got a CVPixelBuffer object. if I does not change the type, I mean pass the same type GL_RGBA, the code could work, but the texture color looks like more blue than normal. After that I change the type like my commit, everything works fine.

Reply:
> Interesting, I wonder if there is a typo in their sample code there?

Apple's documentation just provides "Some example mappings can be seen in the following code snippet."
So there are not all cases. It seems to me that the image type passed should be true order with pixelBuffer, and this func will convert it to GL_RGBA for create texture.

Reply:
I think I'm fine with making the change, though I need to test a few things first to make sure it won't break any of our use-cases.